File: problema 8/c-languaje/problema_8.c
#include <stdio.h>

int main() {
    int matriz[5][5];
    int i, j;
    int sum_pares = 0;
    long long prod_impares = 1;

    // Input elements of the matrix
    printf("Ingrese los elementos de la matriz (5x5):\n");
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            scanf("%d", &matriz[i][j]);
        }
    }

    // Calculate sum of even numbers and product of odd numbers
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            if (matriz[i][j] % 2 == 0) {
                sum_pares += matriz[i][j];
            } else {
                prod_impares *= matriz[i][j];
            }
        }
    }

    // Display the matrix
    printf("Matriz ingresada:\n");
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            printf("%d\t", matriz[i][j]);
        }
        printf("\n");
    }

    // Display results
    printf("Suma de los numeros pares: %d\n", sum_pares);
    printf("Producto de los numeros impares: %lld\n", prod_impares);

    return 0;
}

File: problema 8/fortran/problema_8.f90
program sumAndProduct
    implicit none
    integer :: matriz(5,5)
    integer :: i, j
    integer :: sum_pares
    integer(8) :: prod_impares

    sum_pares = 0
    prod_impares = 1

    ! Input elements of the matrix
    print *, "Ingrese los elementos de la matriz (5x5):"
    do i = 1, 5
        do j = 1, 5
            read *, matriz(i, j)
        end do
    end do

    ! Calculate sum of even numbers and product of odd numbers
    do i = 1, 5
        do j = 1, 5
            if (mod(matriz(i, j), 2) == 0) then
                sum_pares = sum_pares + matriz(i, j)
            else
                prod_impares = prod_impares * matriz(i, j)
            endif
        end do
    end do

    ! Display the matrix
    print *, "Matriz ingresada:"
    do i = 1, 5
        do j = 1, 5
            write(*, '(I3)', advance='no') matriz(i, j)
        end do
        print *, ""
    end do

    ! Display results
    print *, "Suma de los numeros pares: ", sum_pares
    print *, "Producto de los numeros impares: ", prod_impares

end program sumAndProduct